src/libraries/System.Private.CoreLib/src/ILLink/ILLink.LinkAttributes.Shared.xml

```diff
@@ -56,4 +56,46 @@
       <attribute internal="RemoveAttributeInstances" />
     </type>
   </assembly>
+
+  <assembly fullname="System.Private.CoreLib" feature="System.Runtime.InteropServices.BuiltInComInterop.IsSupported" featurevalue="false">
+    <type fullname="System.Runtime.InteropServices.ClassInterfaceAttribute">
```

Comment:
@AaronRobinsonMSFT Could you have a look at these? This instructs IL Linker to remove these custom attributes if built-in COM support is disabled. I assume just stripping some of these attributes will leave us with funny classes that the runtime no longer considers COM, which will result in fun failure modes at runtime ("you want to do COM but you disabled COM" versus "System.Security.SecurityException: 'ECall methods must be packaged into a system module.'").

Reply:
Yes, this will make things fail or work in funny ways. I think that with @LakshanF's recent work to set `g_baseCOMObject` to `NULL` in https://github.com/dotnet/runtime/pull/53928 my concern is slightly less. I will need to consider some of these though.

Reply:
I don't understand this concern. If build-in COM support is disabled such attributes should not be even read by the runtime and if someone would request explicit COM support I'd expect it will get PNSE well before it hits any COM code path.

Reply:
Given sometimes its not easy to see relationship that library components have on built-in COM support, it might be a good idea to do a test run with built-in COM disabled to understand first the impact of this change. We have built-in COM disabled by default for net core applications now and a proactive read is better.

Draft PR #54013 did this for the current state and had a couple of follow ups as can be seen in the PR.

Reply:
> I don't understand this concern.

The concern is that stripping some of these attributes might make the runtime not consider this to be COM anymore and we will miss the "throw COM is disabled message" codepaths.

I don't know nearly enough about COM to point out the specific problems (which is why I asked Aaron for opinion), but if we were to go with what this was originally trying to do, it would transform this:

```csharp
[Guid("CB2F6723-AB3A-11D2-9C40-00C04FA30A3E"), ComImport]
public class CorRuntimeHost
{
}
```

into this:

```csharp
public class CorRuntimeHost
{
    [MethodImpl(InternalCall)]
    public extern void CorRuntimeHost();
}
```

The first one would give you a nice failure mode about COM disabled when the code actually does `new CorRuntimeHost()` at runtime. The second one would throw a weird exception with inscrutable message about how you should package "ECalls" because none of it even hints at COM anymore.

Reply:
>If build-in COM support is disabled such attributes should not be even read by the runtime and if someone would request explicit COM support I'd expect it will get PNSE well before it hits any COM code path.

I think this is a general issue with not built with `FEATURE_COMINTEROP` vs having the feature flag for Com interop set to false. The type loader sets many flags that fundamentally change some behavior &ndash; @MichalStrehovsky's example is spot on.

I will look at each of these. The `Guid` is the worst offender for me but there are others (for example, `CoClassAttribute`) that are incredibly obnoxious.

Reply:
I would remove `CoClassAttribute` and `TypeIdentifierAttribute` from this list. The rest are used during a COM interop scenario not to initiate a scenario.

Reply:
ok, removed them